File: day08/day8/.ts
function findUniqueToy(toy: string): string {
  let counts = new Map()
  for (let letter of toy){
    const key = letter.toLowerCase()
    if (counts.has(key)){
      counts.set(key, {...counts.get(key), count: counts.get(key).count + 1})
    } else {
      counts.set(key, {count: 1, original: letter})
    }
  }
  for (let [_, value] of counts){
    if (value.count == 1) return value.original
  }
  return "" 
}

File: day08/day8/.py
def find_unique_toy(toy: str) -> str:
  counts = {}

  for letter in toy:
    key = letter.lower()
    if key in counts:
      counts[key] = { **counts[key], 'count': counts[key]['count'] + 1 }
    else:
      counts[key] = {'count': 1, 'original': letter}

  for key, value in counts.items():
    if value['count'] == 1: return value['original']
  return ''